### ISSUE TITLE
SpinTransform should propagate Structured inheritance

### ISSUE THREAD

**Issue by arcondello** [dwavesystems/dimod#111]

**Comment by arcondello** [comment]
Another approach would be to have the samplers that require structured children to look down the child-tree.

**Comment by randomir** [comment]
Structure propagation makes more sense to me. That way inconsistencies will be detected earlier (on composition). With the DFS approach from #488, parent structure is ill-defined.

**Comment by arcondello** [comment]
Inheritance has several issues:
* Each composite would need to account for the possibility of having structured children, it would be possible to handle this case in the ABC, but that would be coupling the ComposedSampler and Structured ABCs
* It is not obvious, on a per-composite basis what the right inheritance pattern should be, while #488 is explicit and up to the parent composite that is expecting a structured child to choose.

I agree dfs does not cover all cases.

**Comment by randomir** [comment]
Yes, before clarifying how inheritance should work here, it would be useful to think about composites with multiple children. What would be a good example for such a composite?

**Comment by arcondello** [comment]
I think [dwave-hybrid](https://github.com/dwavesystems/dwave-hybrid) has covered the cases where we would want multiple children, but the original vision for composites included analogues to some hybrid components, for instance `RacingSamplers(sampler0, sampler1)`.
